ardupilot/equipment/power/20008.SmartBatteryContinuous.uavcan

```diff
@@ -0,0 +1,22 @@
+#
+# Smart battery data to be sent continuously (10Hz or greater)
+#
+
+int16 temperature             # [cdegC]
+int16 current                 # [cA]
+uint16 voltage                # [mV]
+int8 battery_remaining          # [%]
```

Comment:
@dakejahl @hendjoshsr71 In MAVLink the main use of `percentage_remaining` is to give the user an intuitive ball park figure of how much battery is left. This level of information is probably all that might be used by some GCS/OSD systems, and is supplied as a "you can just use this without any massaging" figure.

This might be pretty useless for time remaining/charge remaining calculation. That is true whether you use int8, uint8, or represent the percentage as 0-100 or using the full range of the uint 0-254 (with 255 for "not used" indication).

In MAVLink I was thinking of getting the remaining ampere hours (if needed) from the original charge - multiplying [SMART_BATTERY_INFO.capacity_full](https://mavlink.io/en/messages/common.html#SMART_BATTERY_INFO) by the percentage on first use, and then subtracting `current_consumed`

If this is naïve and the value can sensibly be provided, I would certainly consider also having `charge_remaining` in mA hours. The percentage would be expected to match (100*charge_remaining/(charge_remaining+charge_consumed)`

MAVLink is often a bit more considerate of the end consumer than it should be. In any case if it were me I would drop the % in UAVCAN and force MAVLink to calculate it. That is still alignment as long as the mapping is clear. I will then argue the cases in MAVLink for whether % should be sent. 

Does this make sense?

Reply:
PS discussion is continuing on MAVLINk here: https://github.com/mavlink/rfcs/pull/19#discussion_r843415106

Reply:
Yup agreed for UAVCAN I think `charge_remaining` in mAh might be more useful and % remaining can be dropped

Reply:
I might name it `capacity_remaining` since the units of mAh refer to a capacity. Open to other suggestions as well, but current is units of amps... charge is units of coulomb...

Reply:
capacity_remaining makes sense. I would prefer a field with A-h or mA-h if I had to do just one of that or percentage remaining.